containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -362,14 +364,27 @@ class UnorderedMap {
         bitset_type(Kokkos::Impl::append_to_label(prop_copy, " - bitset"),
                     calculate_capacity(capacity_hint));
 
-    m_hash_lists = size_type_view(
-        Kokkos::Impl::append_to_label(prop_copy_noinit, " - hash list"),
-        Impl::find_hash_size(capacity()));
+    if constexpr (alloc_prop_t::sequential_host_init) {
+      m_hash_lists = size_type_view(
+          Kokkos::Impl::append_to_label(prop_copy, " - hash list"),
+          Impl::find_hash_size(capacity()));
 
-    m_next_index = size_type_view(
-        Kokkos::Impl::append_to_label(prop_copy_noinit, " - next index"),
-        capacity() + 1);  // +1 so that the *_at functions can always return a
-                          // valid reference
+      m_next_index = size_type_view(
+          Kokkos::Impl::append_to_label(prop_copy, " - next index"),
+          capacity() + 1);  // +1 so that the *_at functions can always return a
+                            // valid reference
+    } else {
+      const auto prop_copy_noinit = Impl::with_properties_if_unset(
+          prop_copy, Kokkos::WithoutInitializing);
+      m_hash_lists = size_type_view(
+          Kokkos::Impl::append_to_label(prop_copy_noinit, " - hash list"),
+          Impl::find_hash_size(capacity()));
+
+      m_next_index = size_type_view(
+          Kokkos::Impl::append_to_label(prop_copy_noinit, " - next index"),
+          capacity() + 1);  // +1 so that the *_at functions can always return a
+                            // valid reference
+    }
```

Comment:
IMO this is just too much duplication to be acceptable.

Reply:
I tried to redesign it a bit, what do you think?

Reply:
https://github.com/kokkos/kokkos/pull/8182/commits/4753f3b4b1675ee16703f043e9148cba5c8e2eac was fine with me.